satpy/readers/fci_l1c_fdhsi.py

```diff
@@ -131,86 +163,91 @@ def get_channel_dataset(self, channel):
         # Get start/end line and column of loaded swath.
-        self.startline = int(self[measured + "/start_position_row"])
-        self.endline = int(self[measured + "/end_position_row"])
-        self.startcol = int(self[measured + "/start_position_column"])
-        self.endcol = int(self[measured + "/end_position_column"])
-        self.nlines, self.ncols = self[measured + "/effective_radiance/shape"]
+        nlines, ncols = self[measured + "/effective_radiance/shape"]
 
         logger.debug('Channel {} resolution: {}'.format(key.name, chkres))
-        logger.debug('Row/Cols: {} / {}'.format(self.nlines, self.ncols))
-        logger.debug('Start/End row: {} / {}'.format(self.startline, self.endline))
-        logger.debug('Start/End col: {} / {}'.format(self.startcol, self.endcol))
+        logger.debug('Row/Cols: {} / {}'.format(nlines, ncols))
         # total_segments = 70
 
         # Calculate full globe line extent
-        max_y = 5432229.9317116784
-        min_y = -5429229.5285458621
-        full_y = max_y + abs(min_y)
-        # Single swath line extent
-        res_y = full_y / chkres  # Extent per pixel resolution
-        startl = min_y + res_y * self.startline - 0.5 * (res_y)
-        endl = min_y + res_y * self.endline + 0.5 * (res_y)
-        logger.debug('Start / end extent: {} / {}'.format(startl, endl))
-
-        chk_extent = (-5432229.9317116784, endl,
-                      5429229.5285458621, startl)
-        return(chk_extent)
-
-    _fallback_area_def = {
-            "reference_altitude": 35786400,  # metre
-            }
+        h = float(self["data/mtg_geos_projection/attr/perspective_point_height"])
+
+        ext = {}
+        for c in "xy":
+            c_radian = self["data/{:s}/measured/{:s}".format(key.name, c)]
+            c_radian_num = c_radian[:] * c_radian.scale_factor + c_radian.add_offset
+            # FCI defines pixels by centroids (Example Products for Pytroll
+            # Workshop, §B.4.2)
+            #
+            # pyresample defines corners as lower left corner of lower left pixel,
+            # upper right corner of upper right pixel (Martin Raspaud, personal
+            # communication).
+
+            # the .item() call is needed with the h5netcdf backend, see
+            # https://github.com/pytroll/satpy/issues/972#issuecomment-558191583
+            # but we need to compute it first if this is dask
+            min_c_radian = c_radian_num[0] - c_radian.scale_factor.item()/2
+            max_c_radian = c_radian_num[-1] + c_radian.scale_factor.item()/2
+            min_c = min_c_radian * h  # arc length in m
+            max_c = max_c_radian * h
+            try:
+                min_c = min_c.compute()
+                max_c = max_c.compute()
+            except AttributeError:  # not a dask.array
+                pass
+            ext[c] = (min_c.item(), max_c.item())
+
+        area_extent = (ext["x"][1], ext["y"][1], ext["x"][0], ext["y"][0])
```

Comment:
This code assumes that the image is upside down.  Although it has been for publicly available test data, this should not be hardcoded, and this part should be rewritten to be robust in case future versions of test data are not upside down.

Reply:
I think my previous comment is wrong.  I don't think it actually assumes this.